datafusion/expr/src/expr.rs

```diff
@@ -99,21 +99,21 @@ pub enum Expr {
     SimilarTo(Like),
     /// Negation of an expression. The expression's type must be a boolean to make sense.
     Not(Box<Expr>),
-    /// Whether an expression is not Null. This expression is never null.
+    /// True if argument is not NULL, false otherwise. This expression itself is never NULL.
     IsNotNull(Box<Expr>),
-    /// Whether an expression is Null. This expression is never null.
+    /// True if argument is NULL, false otherwise. This expression itself is never NULL.
     IsNull(Box<Expr>),
-    /// Whether an expression is True. Boolean operation
+    /// True if argument is true, false otherwise. This expression itself is never NULL.
     IsTrue(Box<Expr>),
-    /// Whether an expression is False. Boolean operation
+    /// True if argument is  false, false otherwise. This expression itself is never NULL.
     IsFalse(Box<Expr>),
-    /// Whether an expression is Unknown. Boolean operation
+    /// True if argument is NULL, false otherwise. This expression itself is never NULL.
```

Comment:
This might be a little bit misleading, prob we need to add what is unknown ....
https://en.wikipedia.org/wiki/Boolean_data_type#SQL
```
NULL BOOLEAN and UNKNOWN "may be used interchangeably to mean exactly the same thing"
```
so here the user can think that unknown the same as null, but its boolean null, not sure if it matters tbh

Reply:
I think saying these expressions result in NULL is accurate. There is indeed some subtlety related to using expressions that evaluate to `NULL` (booleans) for filters -- where the value does not pass the filter if it evaluates to `false` or `NULL` (it only passes if it is true). 🤯

Reply:
Here's a fun example of that subtlety: https://www.db-fiddle.com/f/xbHAYw59p4tgtR1L39p84x/0